### DISCUSSION TITLE
How can I use an MD5 function in my custom plugin?

### DISCUSSION THREAD

**Discussion by vitorfhc** [Kong/kong#9176] [Help]
I am not a Lua developer, but I have to fix an issue urgently. I've never created a Kong plugin.

It looks like we have [ngx.md5](https://github.com/openresty/lua-nginx-module/tree/v0.10.10#ngxmd5). But, how can I use it?

I tried using it, and I got this: `./access.lua:57: attempt to call field 'md5' (a nil value)`. In the same file, I use things like `ngx.ERROR` and `ngx.log`, so I thought it would work out of the box.

**Comment by flrgh** [comment]
Hi @vitorfhc :wave: 

The kong codebase itself uses `ngx.md5` in many places, so it should indeed be available for you to call. Perhaps there is something else incorrect in the plugin (could be that `ngx` is getting shadowed somewhere). If you can share some example code that reproduces the issue, I'd be happy to give it a look.

Please include the version of Kong that you are using as well.
